src/api/db/models/Project.ts

```diff
@@ -0,0 +1,746 @@
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async updateDeploymentTask(
+    input: gql.UpdateDeploymentInput,
+    context: Context,
+  ): Promise<HydratedDocument<TaskSchema>> {
+    try {
+      return await TaskModel.create(
+        {
+          type: 'UpdateDeployment',
+          projectId: context.user['curr_project'],
+          user: context.user.sub,
+          config: input,
+        },
+        context,
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  // NOTE: this function is called by the task handler
+  static async updateDeployment(
+    input: gql.UpdateDeploymentInput,
+    context: Context,
+  ): Promise<HydratedCameraConfig> {
+    const { cameraId, deploymentId, diffs } = input;
+    try {
+      const { project, camConfig } = await retry(
+        async (bail) => {
+          // find deployment
+          const [project] = await ProjectModel.getProjects(
+            { _ids: [context.user['curr_project']!] },
+            context,
+          );
+          const camConfig = project.cameraConfigs.find((cc) => idMatch(cc._id, cameraId));
+          if (!camConfig) throw new NotFoundError('Camera config not found');
+          const deployment = camConfig!.deployments.find((dep) => idMatch(dep._id!, deploymentId));
+          if (deployment!.name === 'default') {
+            bail(new ForbiddenError(`View ${deployment!.name} is not editable`));
+          }
+
+          // apply updates, sort deployments, and save project
+          Object.assign(deployment!, diffs);
+          camConfig.deployments = sortDeps(camConfig.deployments);
+          await project.save();
+          return { project, camConfig };
+        },
+        { retries: 2 },
+      );
+      // TODO: we need to reverse the above operation if reMapImagesToDeps fails!
+      if (Object.keys(input.diffs).includes('startDate')) {
+        await ProjectModel.reMapImagesToDeps({ projId: project._id, camConfig });
+      }
+      return camConfig;
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async deleteDeploymentTask(
+    input: gql.DeleteDeploymentInput,
+    context: Context,
+  ): Promise<HydratedDocument<TaskSchema>> {
+    try {
+      return await TaskModel.create(
+        {
+          type: 'DeleteDeployment',
+          projectId: context.user['curr_project'],
+          user: context.user.sub,
+          config: input,
+        },
+        context,
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  // NOTE: this function is called by the task handler
+  static async deleteDeployment(
+    { cameraId, deploymentId }: gql.DeleteDeploymentInput,
+    context: Context,
+  ): Promise<HydratedCameraConfig> {
```

Comment:
It looks like we return an object with both the updated `project` and `camConfig` in this method. Also see note about using `<HydratedSingleSubdocument<CameraConfigSchema>>` above.

Reply:
While we do return both a `project` and `camConfig` from the retryable portion of this method, the actual method only returns the `camConfig`:

https://github.com/tnc-ca-geo/animl-api/blob/f9797eb914a6a8662d0fb891f7536a13657fbe89/src/api/db/models/Project.ts#L563-L569

Reply:
Good call!